AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java

```diff
@@ -79,13 +80,24 @@ protected void processParameters(HasPreferences holder, JMethod listenerMethod,
 		JVar newValueParam = listenerMethod.param(classes().OBJECT, "newValue");
 
 		for (VariableElement variableElement : userParameters) {
-			if (variableElement.asType().toString().equals(CanonicalNameConstants.PREFERENCE)) {
+			String type = variableElement.asType().toString();
+
+			if (type.equals(CanonicalNameConstants.PREFERENCE)) {
 				call.arg(preferenceParam);
-			} else if (variableElement.asType().toString().equals(CanonicalNameConstants.OBJECT)) {
+			} else if (type.equals(CanonicalNameConstants.OBJECT)) {
 				call.arg(newValueParam);
+			} else if (type.equals(CanonicalNameConstants.INTEGER) || type.equals(int.class.getName()) || //
+					type.equals(CanonicalNameConstants.FLOAT) || type.equals(float.class.getName()) || //
+					type.equals(CanonicalNameConstants.LONG) || type.equals(long.class.getName())) {
+				JClass wrapperClass = type.startsWith("java") ? holder.refClass(type) : JType.parse(holder.codeModel(), type.replace(".class", "")).boxify();
```

Comment:
The problem is there are two cases:
type is wrapper
type is primitive

In the first case, we can just the holder.refClass. Unfortunetaly in the second case, holder.refClass returns a low letter int class for example. JType.parse is the way to get correct primitive types, but it only expects the name, without the .class. There is no way to get "int" name from "TypeMirror", so we have to remove ".class" manually.